### PR TITLE
Explain that GITHUB_SHA may not always be the commit that was pushed

### LINKED ISSUE #15302: GITHUB_SHA on PRs refers to the merge commit, not the actual commit which triggered the PR
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/learn-github-actions/environment-variables

### What part(s) of the article would you like to see updated?

The docs say that `GITHUB_SHA` is:

> The commit SHA that triggered the workflow. For example, `ffac537e6cbbf934b08745a378932722df287a53`.

This can be a bit misleading for PRs. As best as I can tell, the `GITHUB_SHA` used for PRs is the temporary merge commit GitHub makes against the base branch, i.e. `refs/remotes/pull/<id>/merge`.

When I read the docs, I assumed that "The commit SHA that triggered the workflow" was the SHA of the commit that _I_ pushed to the branch, not the SHA of GitHub's internal merge commit.

I think it would be helpful to mention that the SHA may not be the SHA that you pushed. Not quite sure how to word it, but thought I'd open an issue.

**Why is this important?**

In my case, I was trying to diff the current commit against the upstream branch. Using `GITHUB_SHA` felt like the right handle to use, but didn't work because GITHUB_SHA refers to a merge commit, so there is no diff.

### Additional information

Related issues:

* https://github.com/github/docs/issues/4574
* https://github.com/github/docs/issues/4422

### PR DESCRIPTION
**Pull request by danielcompton**
### Why:

Closes: https://github.com/github/docs/issues/15302

### Check off the following:

- [ ] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Review by Nandojarot — COMMENTED** [review]
Semua telah selesai

**Comment by ethomson** [comment]
Thanks @danielcompton - this makes sense to me.  👍

File: content/actions/learn-github-actions/environment-variables.md [hunk 1]
```diff
@@ -156,7 +156,7 @@ We strongly recommend that actions use environment variables to access the files
 | `GITHUB_RUN_ID` | {% data reusables.github-actions.run_id_description %} For example, `1658821493`. |
 | `GITHUB_RUN_NUMBER` | {% data reusables.github-actions.run_number_description %} For example, `3`. |
 | `GITHUB_SERVER_URL`| The URL of the {% data variables.product.product_name %} server. For example: `https://{% data variables.product.product_url %}`.
-| `GITHUB_SHA` | The commit SHA that triggered the workflow. For example, `ffac537e6cbbf934b08745a378932722df287a53`. |
+| `GITHUB_SHA` | The commit SHA that triggered the workflow. The value of this commit depends on the event that triggered the workflow. For more information, see [Events that trigger workflows](/actions/using-workflows/events-that-trigger-workflows). For example, `ffac537e6cbbf934b08745a378932722df287a53`. |
```
Comment (skedwards88) [inline content/actions/learn-github-actions/environment-variables.md:159]: nit wording change
```suggestion
| `GITHUB_SHA` | The commit SHA that triggered the workflow. The value of this commit SHA depends on the event that triggered the workflow. For more information, see [Events that trigger workflows](/actions/using-workflows/events-that-trigger-workflows). For example, `ffac537e6cbbf934b08745a378932722df287a53`. |
```

**Review by skedwards88 — COMMENTED** [review]
Thanks for this addition! I left a minor wording suggestion that I will go ahead an incorporate, then we'll get this merged down for you.

**Review by skedwards88 — APPROVED** [review]

**Review by Knigt778 — COMMENTED** [review]
M

**Pull request merged**